diff --git a/build/config/posix/BUILD.gn b/build/config/posix/BUILD.gn
index f7dc8c5..1ad313f 100644
--- a/build/config/posix/BUILD.gn
+++ b/build/config/posix/BUILD.gn
@@ -39,17 +39,38 @@ config("runtime_library") {
     asmflags += sysroot_flags
     ldflags += sysroot_flags
 
+    _multiarch_include = ""
+    if (is_linux || is_chromeos) {
+      if (current_cpu == "x64") {
+        _multiarch_include = "x86_64-linux-gnu"
+      } else if (current_cpu == "loong64") {
+        _multiarch_include = "loongarch64-linux-gnu"
+      } else if (current_cpu == "riscv64") {
+        _multiarch_include = "riscv64-linux-gnu"
+      }
+    }
+
+    sysroot_include_flags = []
+    if (_multiarch_include != "" &&
+        path_exists("$sysroot/usr/include/$_multiarch_include")) {
+      sysroot_include_flags = [
+        "-isystem" +
+            rebase_path("$sysroot/usr/include/$_multiarch_include",
+                        root_build_dir),
+      ]
+    }
+
     # When use_custom_libcxx=true, some -isystem flags get passed to
     # cflags_cc to set up libc++ include paths.  We want to make sure
     # the sysroot includes take lower precendence than the libc++
     # ones, so they must appear later in the command line.  However,
     # the gn reference states "These variant-specific versions of
     # cflags* will be appended on the compiler command line after
     # 'cflags'."  Because of this, we must set the sysroot flags for
     # all cflags variants instead of using 'cflags' directly.
-    cflags_c += sysroot_flags
-    cflags_cc += sysroot_flags
-    cflags_objc += sysroot_flags
-    cflags_objcc += sysroot_flags
+    cflags_c += sysroot_flags + sysroot_include_flags
+    cflags_cc += sysroot_flags + sysroot_include_flags
+    cflags_objc += sysroot_flags + sysroot_include_flags
+    cflags_objcc += sysroot_flags + sysroot_include_flags
   }
 }
