diff --git a/src/string.rs b/src/string.rs
index 18c5ece..c1dc0d5 100644
--- a/src/string.rs
+++ b/src/string.rs
@@ -446,8 +446,8 @@ bitflags! {
   #[derive(Clone, Copy, Default)]
   #[repr(transparent)]
   pub struct WriteFlags: int {
-    const kNullTerminate = crate::binding::v8_String_WriteFlags_kNullTerminate as _;
-    const kReplaceInvalidUtf8 = crate::binding::v8_String_WriteFlags_kReplaceInvalidUtf8 as _;
+    const kNullTerminate = crate::binding::WriteFlags_kNullTerminate as _;
+    const kReplaceInvalidUtf8 = crate::binding::WriteFlags_kReplaceInvalidUtf8 as _;
   }
 }
